base/src/main/java/org/aya/core/pat/PatMatcher.java

```diff
@@ -77,23 +71,22 @@ private void match(@NotNull Pat pat, @NotNull Term term) throws Mismatch {
         }
       }
       case Pat.Tuple tuple -> {
+        term = pre.apply(term);
         switch (term) {
           case IntroTerm.Tuple tup -> visitList(tuple.pats(), tup.items());
           case RefTerm.MetaPat metaPat -> solve(pat, metaPat);
           default -> throw new Mismatch(true);
         }
       }
-      case Pat.Meta meta -> {
-        var sol = meta.solution().get();
-        assert sol != null : "Unsolved pattern " + meta;
-        match(sol, term);
-      }
+      case Pat.Meta ignored -> throw new InternalException("Pat.Meta is not allowed");
```

Comment:
Why we do not match the solution against the term? I guess we inlined these solutions before the match?

Reply:
I think we should inline all `Pat.Meta` before it can be used to match, so any `Pat.Meta` implies something was not inlined which is a bug.